crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -189,17 +190,20 @@ pub async fn app_validation_workflow(
         trigger_integration.trigger(&"app_validation_workflow");
     }
 
+    let validations_dependencies = validation_dependencies.lock();
     Ok(
         // If not all ops have been validated
         // and fetching missing hashes has not timed out,
         // trigger app validation workflow again.
         if outcome_summary.validated < outcome_summary.ops_to_validate
-            && !validation_dependencies
-                .lock()
-                .fetch_missing_hashes_timed_out()
+            && !validations_dependencies.fetch_missing_hashes_timed_out()
         {
-            // Trigger app validation workflow again in 10 seconds.
-            WorkComplete::Incomplete(Some(Duration::from_secs(10)))
+            // Trigger app validation workflow again in 100-1000 milliseconds.
+            let interval = max(
+                1000 - validations_dependencies.missing_hashes.len() * 100,
+                100,
+            );
```

Comment:
since `len` is usize, isn't this going to underflow at 11? Should we rather saturate?

```suggestion
            let interval = 1000.saturating_sub(validations_dependencies.missing_hashes.len() * 100);
```

Reply:
Could well be that it's overflowing. `1000 - 11usize * 100` doesn't even compile. I take your offer.